src/ResourceManager/Search/Commands.Management.Search/SearchService/GetSearchServiceQueryKeyCommand.cs

```diff
@@ -0,0 +1,81 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Management.Search.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using System;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Management.Search.SearchService
+{
+    [Cmdlet(VerbsCommon.Get, SearchServiceQueryKeyNounStr), OutputType(typeof(PSSearchQueryKey))]
+    public class GetSearchServiceQueryKeyCommand : SearchServiceBaseCmdlet
+    {
+        [Parameter(
+         Position = 0,
+         Mandatory = true,
+         ValueFromPipeline = true,
+         ParameterSetName = ParentObjectParameterSetName,
+         HelpMessage = InputObjectHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public PSSearchService ParentObject { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = ParentResourceIdParameterSetName,
+            HelpMessage = ResourceIdHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string ParentResourceId { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
```

Comment:
ValueFromPipelineByProperty should only ever be applied to ResourceId/ParentResourceId, and ValueFromPipeline should only be applied to InputObject/ParentObject.  Fix everywhere.